### PR TITLE
Support TmpFS mappings

### PR DESCRIPTION
**Pull request by malltshik**
See #673

### CODE REVIEW

File: core/src/test/java/org/testcontainers/junit/GenericContainerRuleTest.java [hunk 1]
```diff
@@ -158,6 +160,28 @@ public void testIsRunning() {
...
+            // check file doesn't exist
+            String path = "/testtmpfs/test.file";
+            Container.ExecResult execResult = container.execInContainer("ls", path);
+            assertEquals("tmpfs inside container works fine", execResult.getStderr(),
+                "ls: /testtmpfs/test.file: No such file or directory\n");
+            // touch && check file does exist
+            container.execInContainer("touch", path);
+            execResult = container.execInContainer("ls", path);
+            assertEquals("tmpfs inside container works fine", execResult.getStdout(), path + "\n");
+        } catch (InterruptedException | IOException ignored) {
```
Comment (bsideup) [inline core/src/test/java/org/testcontainers/junit/GenericContainerRuleTest.java]: this catch is not needed, just add `throws Exception` to the test method

File: core/src/test/java/org/testcontainers/junit/GenericContainerRuleTest.java [hunk 2]
```diff
@@ -158,6 +160,28 @@ public void testIsRunning() {
         }
     }
 
+    @Test
+    public void withTmpFsTest(){
+        try(GenericContainer container = new GenericContainer().withCommand("top")
```
Comment (bsideup) [inline core/src/test/java/org/testcontainers/junit/GenericContainerRuleTest.java]: please see other tests on how to format it, something like:
```java
try (
    GenericContainer container = new GenericContainer()
                                             .withCommand("top")
                                             .withTmpFs(singletonMap("/testtmpfs", "rw"))
) {
```

File: core/src/test/java/org/testcontainers/junit/GenericContainerRuleTest.java [hunk 3]
```diff
@@ -158,6 +160,28 @@ public void testIsRunning() {
...
+            .withTmpFs(singletonMap("/testtmpfs", "rw"))) {
+            assertFalse("Container is not started and not running", container.isRunning());
+            container.start();
+            assertTrue("Container is started and running", container.isRunning());
```
Comment (bsideup) [inline core/src/test/java/org/testcontainers/junit/GenericContainerRuleTest.java]: this test only tests TmpFS, no need to assert container's "running" state

File: core/src/test/java/org/testcontainers/junit/GenericContainerRuleTest.java [hunk 4]
```diff
@@ -158,6 +160,28 @@ public void testIsRunning() {
         }
     }
 
+    @Test
+    public void withTmpFsTest(){
```
Comment (bsideup) [inline core/src/test/java/org/testcontainers/junit/GenericContainerRuleTest.java]:
```suggestion
     public void withTmpFsTest() {
```

File: gradle.properties [hunk 5]
```diff
@@ -1,2 +1,3 @@
 org.gradle.parallel=false
 org.gradle.caching=true
+org.gradle.java.home=/usr/lib/jvm/java-8-openjdk-amd64/
```
[the remark this answers is not in the document]
Comment (malltshik) [reply gradle.properties]: Yes indeed. My fault. I reset whole changes linked with gradle.

**Comment by rnorth** [comment]
Thanks @malltshik! Will merge now.

**Comment by malltshik** [comment]
Thank you for your time. That was great practice for me.

**Comment by rnorth** [comment]
Released in [1.10.4](https://github.com/testcontainers/testcontainers-java/releases/tag/1.10.4)!

Thanks for the contribution!

**Pull request merged**